src/lib.rs

```diff
@@ -144,6 +150,67 @@ where
         }
     }
 
+    pub fn read_co2_ppm_and_temp_celcius(&mut self) -> nb::Result<(u16, f32), Error<E>> {
```

Comment:
This function should have documentation stating that firmware version >= 5 (i.e. `0500`) is required.

Reply:
Well, I guess so

But that does not feel the rust way to let the user shoot themselves in the foot.
So suddenly I had an idea that if we could have 2 (possibly later even more states) where a user of the libary first had to do something like
`mhz19c_upgraded = mhz19c.upgrade_capabilities().unwrap()` where the upgrade_capabilities does the FW check, and if it fails it would return an error; and if it succeeeds it would return an new MhZ19c which has the same functions + some extra functions (in this case this one).

So that would mean that we can be certain on compile time you are not going to call this function if your hardware does not allow it.

I just have not really a clue how to program this in Rusts type system (I am certain it must be possible, just not how...)

Or is this overkill?

Reply:
I like that idea and took a stab at it based on your code. I at least got it it to compile and made a PR against your branch: nathansamson/mh-z19c#1

I still have to test it though. (I'm not completely sure how ergonomic this is for the consumer of the API.)